lib/datadog/appsec/contrib/sinatra/patcher.rb

```diff
@@ -106,7 +107,7 @@ def process_route(*)
                 self.response = AppSec::Response.negotiate(env).to_sinatra_response
 
                 # interrupt request and return response to dispatch! for consistency
-                throw(Ext::ROUTE_INTERRUPT, response)
+                throw(Datadog::AppSec::Contrib::Sinatra::Ext::ROUTE_INTERRUPT, response)
```

Comment:
I found out about this 🐛 while working on removing all the unused code 😄 

Thank god it hasn't reached production 😌

Reply:
From a cursory look I don't see the bug:

Ruby would try to resolve `Ext::ROUTE_INTERRUPT` this way:

- `Ext::ROUTE_INTERRUPT` starts with `Ext`, so try to resolve that:
  - look for `Ext` in each component of `Datadog::AppSec::Contrib::Sinatra::Patcher`, popping each until `Ext` is found
    - `Ext` not in `Datadog::AppSec::Contrib::Sinatra::Patcher` => move on
    - `Ext` in `Datadog::AppSec::Contrib::Sinatra` => use that
- `ROUTE_INTERRUPT` in `Datadog::AppSec::Contrib::Sinatra::Ext`, so use that
- constant resolved!

```
module Foo
  module Bar
    module Baz
      def self.test
        X::Y
      end
    end

    module X
      Y = 'y'
    end
  end
end

p Foo::Bar::Baz.test # => "y"
```

Not that I don't condone the change, because the moment you introduce `Ext` at the `Patcher` level for some reason it would find that `Ext` instead and blow up:

```
module Foo
  module Bar
    module Baz
      def self.test
        X::Y
      end

      module X; end
    end

    module X
      Y = 'y'
    end
  end
end

p Foo::Bar::Baz.test # => NameError
```